platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/cleanup/ProgramExecutionsCleaner.java

```diff
@@ -0,0 +1,37 @@
+package com.softicar.platform.core.module.program.execution.cleanup;
+
+import com.softicar.platform.common.date.DayTime;
+import com.softicar.platform.core.module.CoreI18n;
+import com.softicar.platform.core.module.event.SystemEventBuilder;
+import com.softicar.platform.core.module.event.severity.AGSystemEventSeverityEnum;
+import com.softicar.platform.core.module.program.Programs;
+import com.softicar.platform.core.module.program.execution.AGProgramExecution;
+import com.softicar.platform.core.module.program.state.AGProgramState;
+import com.softicar.platform.db.core.transaction.DbTransaction;
+
+public class ProgramExecutionsCleaner {
+
+	public void cleanupOrphanedRecords() {
+
+		try (var transaction = new DbTransaction()) {
+			AGProgramExecution.TABLE//
+				.createSelect()
+				.where(AGProgramExecution.TERMINATED_AT.isNull())
+				.joinLeftReverse(AGProgramState.CURRENT_EXECUTION)
+				.where(AGProgramState.TABLE.isNull())
+				.forEach(this::terminateExecutionAndCreateEvent);
+			transaction.commit();
+		}
+	}
+
+	private void terminateExecutionAndCreateEvent(AGProgramExecution execution) {
+
+		execution//
+			.setFailed(true)
+			.setTerminatedAt(DayTime.now())
+			.save();
+		new SystemEventBuilder(AGSystemEventSeverityEnum.ERROR, CoreI18n.PROGRAM_EXECUTION_FAILED.toString())//
```

Comment:
I wonder if this shouldn't be rather `PROGRAM_EXECUTION_WAS_ABORTED` or similar?

Reply:
Hm that implies that all orphaned executions were aborted, dont know if that is correct.
Failed is imho a good term, because a execution had to be cleaned, therefore it failed in some way

Reply:
actually, if the execution record remained, that can only mean that someone killed the JVM, so the program was actually aborted. In my book, "fail" means there was some kind of error in the program execution, with a high likeliness of reoccurring, which is here not given.

but if you insist, this will not stop me to accept

Reply:
maybe we should even be more explicit and write something like ORPHANED_PROGRAM_EXECUTION, so the system admin may remember: "Ah these executions where kill the yesterday, when I restarted the server."

Reply:
oh thats a good idea yeah

Reply:
also changed system event to be a warning, because if it really only happends when the JVM gets restartet, its not exactly an error in my book

Reply:
ok